systemservices/marketplace/src/contracts/version.ts

```diff
@@ -12,7 +12,8 @@ const getServiceVersions = async (contract: Marketplace, sid: string): Promise<V
   const versionsLength = new BigNumber(await contract.methods.serviceVersionsLength(sidHex).call())
   const versionsPromise: Promise<Version|undefined>[] = []
   for (let j = new BigNumber(0); versionsLength.isGreaterThan(j); j = j.plus(1)) {
-    versionsPromise.push(getServiceVersionWithIndex(contract, sid, j))
+    const version = getServiceVersionWithIndex(contract, sid, j)
+    if (version) versionsPromise.push(version)
```

Comment:
This is useless because it returns a promise, and 2 lines under, empty elements are removed from array

Reply:
done